move_period_from_transaction_date/model/account_move_line.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+"""Extend account.move.line to use date for period, when requested."""
+##############################################################################
+#
+#    Copyright (C) 2015 Therp BV - http://therp.nl.
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp import models, api
+
+
+class AccountMoveLine(models.Model):
+    """Extend account.move.line to use date for period, when requested."""
+    _inherit = 'account.move.line'
+
+    @api.model
+    def create(self, vals, check=True):
```

Comment:
Isn't better to override `_prepare_move_line_vals` in account.bank.statement to not interfere in this frequently called method?